internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go

```diff
@@ -902,27 +981,53 @@ func expandClusterProperties(model *ClusterResourceModel) *managedcluster.Manage
 				Protocol:         rule.Protocol,
 			}
 
-			fePortStr := strconv.FormatInt(rule.FrontendPort, 10)
-			var sgProto managedcluster.NsgProtocol
+			fePortStr = strconv.FormatInt(rule.FrontendPort, 10)
 			switch rule.Protocol {
 			case managedcluster.ProtocolTcp:
 				sgProto = managedcluster.NsgProtocolTcp
 			case managedcluster.ProtocolUdp:
 				sgProto = managedcluster.NsgProtocolUdp
 			}
-			nsRules[idx] = managedcluster.NetworkSecurityRule{
-				Access:                     managedcluster.AccessAllow,
-				SourceAddressPrefixes:      &[]string{"0.0.0.0/0"},
-				SourcePortRanges:           &[]string{"1-65535"},
-				DestinationPortRanges:      &[]string{fePortStr},
-				DestinationAddressPrefixes: &[]string{"0.0.0.0/0"},
-				Direction:                  managedcluster.DirectionInbound,
-				Name:                       fmt.Sprintf("rule%d-allow-fe", rule.FrontendPort),
-				Priority:                   1000,
-				Protocol:                   sgProto,
+			frontendPort = rule.FrontendPort
+
+			if ns := model.NSRules; len(ns) == 0 {
+				nsRules[idx] = managedcluster.NetworkSecurityRule{
+					Access:                     managedcluster.AccessAllow,
+					SourceAddressPrefixes:      &[]string{"0.0.0.0/0"},
+					SourcePortRanges:           &[]string{"1-65535"},
+					DestinationPortRanges:      &[]string{fePortStr},
+					DestinationAddressPrefixes: &[]string{"0.0.0.0/0"},
+					Direction:                  managedcluster.DirectionInbound,
+					Name:                       fmt.Sprintf("rule%d-allow-fe", frontendPort),
+					Priority:                   1000 + int64(idx),
```

Comment:
these need to be user configurable

Reply:
@tombuildsstuff thanks for your reply. Yes, the Priority has been exposed to the user as follows. This is the default value that prevents breaking change when not set by the user.
![image](https://user-images.githubusercontent.com/39109137/196314651-c6fc73ad-282a-4188-b0fe-559343816d74.png)

Reply:
those are network security rules, not load balancer rules? to fix this we need to make it uer configurable, and then default them to these only values (only for lb rules) - likely in schema. maybe i'm missing something but you've not made all these properties setable?